src/main/java/com/google/devtools/build/lib/skyframe/TreeArtifactValue.java

```diff
@@ -190,23 +192,35 @@ public static ArchivedRepresentation create(
    */
   @Nullable private final ArchivedRepresentation archivedRepresentation;
 
+  /**
+   * Optional symlink target path.
+   *
+   * <p>If set, this artifact is a copy of another artifact. It is materialized in the local
+   * filesystem as a symlink to the original artifact, whose contents live at this location. This is
+   * used to implement zero-cost copies of remotely stored artifacts.
+   */
+  @Nullable private final PathFragment symlinkTargetExecPath;
```

Comment:
I think you measured the memory impact for adding this new field. I'm curious about the numbers.

Reply:
In the end, I didn't benchmark; I just made sure that I'm not increasing the size of `TreeArtifactValue` nor `RemoteFileArtifactValue`, which means the change will be neutral for builds that don't use `ctx.actions.symlink(output = declare_file/declare_directory)`.

(I think there's a way to make it truly neutral by using different specializations of `RemoteFileArtifactValue` for `actionId` and `locationIndex`, since AIUI their presence is mutually exclusive. But I'd rather avoid doing that now if I can.)

Reply:
From our offline discussion, it looks like we might be able to get rid of `actionId` entirely, so that would be preferred.